### DISCUSSION TITLE
Kubernetes replicas

### DISCUSSION THREAD

**Discussion by raffaele-pilloni** [plausible/analytics#1971] [Self-Hosted Support]
Hi Plausible Community.

In the deployment.yaml file it says that it is preferable not to change the replicas parameter but I have seen that many examples do.

[deployment.yaml](https://github.com/plausible/hosting/blob/master/kubernetes/plausible.yaml)

 i wanted to know if plausible can work on kubernetes with number of replicas greater than one?

**Comment by schnerring** [comment]
See also: https://github.com/plausible/hosting/pull/3#issuecomment-842114779

> Plausible will not work in a clustered scenario. The session state is kept in memory and if you start load balancing between multiple nodes, the session state will be fragmented. This limits Plausible to a single node at the moment.

**Reply by raffaele-pilloni** [reply]
For now i solved with sticky-session. but I think it might lose some information during the autoscaling or some rollout-restart.

**Comment by cnkk** [comment]
I think this information is outdated now.
As long as a request from an IP/session is always redirected to the same backend application, clustering the plausible container should work.
In the k8s context you could use for example envoy or even istio to load balance traffic based on a cookie or HTTP header.

**Reply by raffaele-pilloni** [reply]
I already do a similar thing with sticky-session but during the autoscaling or some rollout-restart some information might be lost.
